pkg/apis/migration/v1alpha1/gvk.go

```diff
@@ -44,3 +50,72 @@ func (i *Incompatible) ResourceList() (incompatible []string) {
 	}
 	return
 }
+
+func CollectResources(discovery discovery.DiscoveryInterface) ([]*metav1.APIResourceList, error) {
```

Comment:
`CollectResources` and `ConvertToGVRList` are reusable, so they were moved here.